sales_team_security/models/ir_rule.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2016-2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models, tools
+from odoo.osv import expression
+from odoo.tools import config
+
+
+class IrRule(models.Model):
+    _inherit = 'ir.rule'
+
+    @api.model
+    @tools.conditional(
+        'xml' not in config['dev_mode'],
+        tools.ormcache(
+            'self._uid', 'model_name', 'mode',
+            'tuple(self._context.get(k) for k in self._compute_domain_keys())'
+        )
+    )
+    def _compute_domain(self, model_name, mode="read"):
+        """Inject extra domain for restricting partners when the user
+        has the group 'Sales / User: Own Documents Only'.
+        """
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        group1 = "sales_team.group_sale_salesman"
+        group2 = "sales_team_security.group_sale_team_manager"
+        group3 = "sales_team.group_sale_salesman_all_leads"
+        if model_name == "res.partner":
```

Comment:
:thinking: If you're affecting only the `res.partner` model, don't you think `res.partner.check_access_rule()` would be the appropriate hook to override?

Reply:
That method only serves for direct operations over the record, but for example it doesn't limit the number or records returned by a `search`. But this low level method is call for all of them.

Reply:
Interesting, today I learned that.